distributed/deploy/tests/test_adaptive.py

```diff
@@ -460,3 +460,19 @@ async def test_update_adaptive(cleanup):
         await asyncio.sleep(0.2)
         assert first.periodic_callback is None
         assert second.periodic_callback.is_running()
+
+
+@pytest.mark.asyncio
+async def test_adaptive_no_memory_limit(cleanup):
+    """Make sure that adapt() does not keep creating workers when no memory limit is set."""
+    async with LocalCluster(
+        n_workers=0, threads_per_worker=1, memory_limit=0, asynchronous=True
+    ) as cluster:
+        cluster.adapt(minimum=1, maximum=10, interval="100 ms")
+        async with Client(cluster, asynchronous=True) as client:
+            await client.gather(client.map(sleep, range(3)))
+        assert all(
+            state[1]["n"] <= 3
+            for state in cluster._adaptive.log
+            if state[1]["status"] == "up"
+        )
```

Comment:
Test runtime is not a huge concern. By using `slowinc` instead of sleep we can reduce runtime down to about 2s. That's effectively just a smaller sleep interval.

One thing I'd like to get in, though, is to change the assert statement. You are testing that there is no individual scaling beyond three instance but I think we should rather assert that there are not more than three in total. 

```suggestion
            await client.gather(client.map(slowinc, range(3)))
        assert (
            sum(
                state[1]["n"]
                for state in cluster._adaptive.log
                if state[1]["status"] == "up"
            )
            <= 3
        )
```

Reply:
Thank you! I will locally test `slowinc`. I think I had tried it at some point, but then the test did not fail when I removed the other parts of this PR. Probably needs some fine-tuning of `interval="100 ms"` and the number of `tasks`.

Reply:
you can also tweak the sleep time by calling `client.map(sleep, range(3), seconds=0.05)`. Same for `incslow` but the argument is called differently

Reply:
Thanks! I assume I will need to also use the git version of dask and not 2021.04.0 as dask seems to import a non-exiting function:
```py
pytest -k test_adaptive_no_memory_limit distributed/deploy/tests/test_adaptive.py
[...]
>       from distributed.protocol.core import dumps_msgpack
E       ImportError: cannot import name 'dumps_msgpack' from 'distributed.protocol.core' (/results/twoertwe/home/git/distributed/distributed/protocol/core.py)

../../.cache/pypoetry/virtualenvs/python-tools-0mbQS0_g-py3.9/lib/python3.9/site-packages/dask/highlevelgraph.py:946: ImportError
```

Reply:
Yes, this was breaking in dask. We'll have a release upcoming this week, I believe which should resolve this. For now, you'll need master :/

Reply:
I wasn't able to bring the `sleep`/`slowinc` below 0.35s, but now it fails when the actual patch isn't applied :)

Reply:
@fjetter is there anything more to do?